### PR TITLE
Xavier nx fxtof enum

### PR DESCRIPTION
**Pull request by rbudai98**

### CODE REVIEW

File: sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp [hunk 1]
```diff
@@ -44,16 +46,115 @@
...
+    char cmdDev[96];
+    sprintf(cmdDev, "v4l2-ctl --list-devices | grep addi9036 -A 2 | sed '1d'");
+    FILE *fp = popen(cmdDev, "r");
+    if (!fp) {
+        LOG(WARNING) << "Error running v4l2-ctl";
+        return Status::GENERIC_ERROR;
+    }
+
+    /* Read the media-ctl output stream */
+    buf = (char *)malloc(128 * 1024);
```
Comment (dNechita) [inline sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp:66]: Not freed anywhere

File: sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp [hunk 2]
```diff
@@ -44,16 +46,115 @@
...
+    char cmdSubDev[2500];
+    sprintf(cmdSubDev, "media-ctl -p");
+    FILE *subDevFp = popen(cmdSubDev, "r");
+    if (!subDevFp) {
+        LOG(WARNING) << "Error running media-ctl";
+        return Status::GENERIC_ERROR;
+    }
+
+    /* Read the media-ctl output stream */
+    subDevBuf = (char *)malloc(2500 * sizeof(char));
```
Comment (dNechita) [inline sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp:103]: Not freed anywhere

File: sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp [hunk 3]
```diff
@@ -44,16 +46,115 @@
...
+    }
+    return Status::OK;
+}
+
 Status TargetSensorEnumerator::searchSensors() {
 
+    Status status = Status::OK;
     LOG(INFO) << "Looking for devices on the target: Xavier NX";
 
     // TO DO: Don't guess the device, find a way to identify it so we are sure
```
Comment (dNechita) [inline sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp:51]: Remove comment please

File: sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp (hunk 1 above)
Comment (rbudai98) [reply sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp:66]: fixed

File: sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp (hunk 2 above)
Comment (rbudai98) [reply sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp:104]: fixed

File: sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp (hunk 3 above)
Comment (rbudai98) [reply sdk/src/cameras/ad-fxtof1-ebz/xavier_nx/sensor_enumerator_xaviernx.cpp:51]: fixed

**Review by dNechita — APPROVED** [review]

**Pull request merged**